compiler/test/dotty/tools/vulpix/ParallelTesting.scala

```diff
@@ -1088,7 +1088,10 @@ trait ParallelTesting extends RunnerOrchestration { self =>
     targetDir.mkdirs()
     assert(targetDir.exists, s"couldn't create target directory: $targetDir")
 
-    val target = JointCompilationSource(s"$testName from $callingMethod", files.map(new JFile(_)).toArray, flags, targetDir)
+    // Sort the files so that the compilation order is deterministic.
+    // See https://github.com/lampepfl/dotty/issues/2792
+    val sorted = files.map(new JFile(_)).toArray.sorted
```

Comment:
I saw it, but it seemed a bit premature to add a seed argument to `compileList` that isn't currently used by any of the callers. If somebody needs it down the line, we can always refactor then. I don't feel strongly about this, so happy to change it if you want.